### PR TITLE
browser(webkit): instrument event source

### PR DESCRIPTION
**Pull request by aslushnikov**
Review URL: https://github.com/aslushnikov/webkit/commit/721d999f1034771b3765ba70fa3942cdb33acf27

References #2189

### CODE REVIEW

**Review by dgozman — APPROVED** [review]

File: browser_patches/webkit/patches/bootstrap.diff [hunk 1]
```diff
@@ -2201,15 +2216,23 @@ index 03bdd55cc930de0cfc61f8d9e4210f4952aa4633..cd95112f4ed24429dfe27343228592e1
      static void willComposite(Frame&);
      static void didComposite(Frame&);
      static void willPaint(RenderObject&);
-@@ -198,6 +205,7 @@ public:
+@@ -191,6 +198,7 @@ public:
+     static void applyUserAgentOverride(Frame&, String&);
+     static void applyEmulatedMedia(Frame&, String&);
+ 
++
```
Comment (yury-s) [inline browser_patches/webkit/patches/bootstrap.diff:2223]: revert this line to minimize diff?

File: browser_patches/webkit/patches/bootstrap.diff [hunk 2]
```diff
@@ -2138,11 +2147,17 @@ index 0c442ee908877d2067d65e87441652a8cf00c88e..6fb448784e0d97fe60eb233853905138
 +        return networkAgent->interceptRequest(loader, WTFMove(handler));
 +    return false;
 +}
++
++void InspectorInstrumentation::willSendEventSourceRequestImpl(InstrumentingAgents& instrumentingAgents)
++{
++    if (InspectorNetworkAgent* networkAgent = instrumentingAgents.inspectorNetworkAgent())
++        networkAgent->willSendEventSourceRequest();
```
Comment (yury-s) [inline browser_patches/webkit/patches/bootstrap.diff:2154]: This is brittle, what if willSendRequest is not called synchronously right after this? Maybe there is a way to pass initiator from the loader that creates request?

**Review by yury-s — APPROVED** [review]

File: browser_patches/webkit/patches/bootstrap.diff [hunk 3]
```diff
@@ -5046,6 +5111,26 @@ index 6ffc83c012ffe095046414051161d7506af192a1..8b5095998153f0c3a5762022459dbd09
...
+ #include "EventNames.h"
++#include "InspectorInstrumentation.h"
+ #include "MessageEvent.h"
+ #include "ResourceError.h"
+ #include "ResourceRequest.h"
+@@ -93,6 +94,7 @@ EventSource::~EventSource()
+ 
+ void EventSource::connect()
+ {
++    InspectorInstrumentation::willSendEventSourceRequest(scriptExecutionContext());
```
Comment (yury-s) [inline browser_patches/webkit/patches/bootstrap.diff:5130]: Can we instead set `inspectorInitiatorNodeIdentifier` or `inspectorInitiatorIdentifier` on the request created a few lines below and then check it inside the inspector?

**Comment by aslushnikov** [comment]
Closing in favor of #2423

**Pull request closed without merging**